### PR TITLE
[ticket/12610] Add command to check if the board is up to date.

### PR DESCRIPTION
**Pull request by Nicofuma**
PHPBB3-12610

### CODE REVIEW

**Comment by Nicofuma** [comment]
replaces https://github.com/phpbb/phpbb/pull/2540

**Comment by Nicofuma** [comment]
to be done: long help

@VSEphpbb ping? =)

**Comment by marc1706** [comment]
The tests fail because version check is now ran by default on all extensions in the list. This causes the two actual extensions to output errors because they don't have any version check info inside them. I'm not sure if this is behavior we should want by default?
@paul999 @VSEphpbb @Derky

**Comment by paul999** [comment]
Can't we skip extension which doesn't have any version check information? Having the command is usefull, especially for extensions as they are updated more often.

**Comment by Derky** [comment]
The version check info in composer is optional so it shouldn't error when an extension doesn't come with a version check URL.

**Comment by marc1706** [comment]
I'd recommend to let this fail silently on the extensions page.

**Comment by paul999** [comment]
Or maybe just display something like N/A?

**Comment by marc1706** [comment]
This already happens on the overview page. There is no update info displayed on that page. It's rather just green/red or black for no info.

**Comment by paul999** [comment]
I would leave it black in that case.

**Comment by marc1706** [comment]
It seems like right now this does not properly check for the latest version on the extensions overview page. One needs to click on details to have phpBB check if extensions are up to date.

Before that, error messages like this one are displayed:
```
The “forumhulp/realpostcount” extension is not valid.
Failed to obtain latest version information.
```
This is obviously incorrect as this message disappears once one clicks on the extension details.

**Comment by marc1706** [comment]
@Nicofuma what's the status of this?

**Comment by Nicofuma** [comment]
kinda forgot about it. Will update the branch soon

**Comment by marc1706** [comment]
Status?

**Comment by Nicofuma** [comment]
well... I just need time to finish it (I need to not forget about it too...)

**Comment by Nicofuma** [comment]
rebased and fixed

File: phpBB/bin/phpbbcli.php [hunk 1]
```diff
@@ -42,6 +42,7 @@
 require($phpbb_root_path . 'includes/functions.' . $phpEx);
 require($phpbb_root_path . 'includes/functions_admin.' . $phpEx);
 require($phpbb_root_path . 'includes/utf/utf_tools.' . $phpEx);
+include($phpbb_root_path . 'includes/functions_compatibility.' . $phpEx);
```
Comment (marc1706) [inline phpBB/bin/phpbbcli.php]: Only conflict right now is that this was already added by another pull request.

File: phpBB/phpbb/console/command/update/check.php [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+			{
+				$this->display_versions($output, $updates_available);
+			}
+
+			return 1;
+		}
+		else
+		{
+			$output->writeln('');
+			$output->writeln('<question>' . $this->user->lang('NOT_UP_TO_DATE', $metadata['name']) . '</question>');
```
Comment (marc1706) [inline phpBB/phpbb/console/command/update/check.php]: Looks like a copy paste typo. Should display that it's up to date.

File: phpBB/phpbb/console/command/update/check.php [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+			}
+
+			return 1;
+		}
+		else
+		{
+			if ($input->getOption('verbose'))
+			{
+				$output->writeln('');
+				$output->writeln('<question>' . $this->user->lang('UPDATE_NOT_NEEDED') . '</question>');
```
Comment (marc1706) [inline phpBB/phpbb/console/command/update/check.php]: I think this should always be displayed, even if verbose is disabled. Right now this just returns empty.

**Comment by marc1706** [comment]
Don't you think it might be good to use the SymfonyStyle I/O styling that symfony introduced in 2.8?
http://symfony.com/blog/new-in-symfony-2-8-console-style-guide

**Comment by VSEphpbb** [comment]
^ What @marc1706 said :)

**Comment by marc1706** [comment]
Also, it looks like you're still using `$user->lang()` instead of the new language class. I guess that's still from when you initially based this for 3.1.x?

**Comment by marc1706** [comment]
I think you should also catch the exception and do a "nice" output when someone tries to check a non-existent extension for updates:
```
[RuntimeException]                                                                                       
  The requested file could not be found: phpBB/bin/../ext/foobar/composer.json                                                                                                         
```

**Comment by Nicofuma** [comment]
right, I'll do that later

**Comment by Nicofuma** [comment]
@marc1706 updated

**Comment by marc1706** [comment]
Looks like you forgot to update the tests.

**Pull request merged**